### DISCUSSION TITLE
How to log one to many activity?

### DISCUSSION THREAD

**Discussion by airliajsmith** [spatie/laravel-activitylog#792] [Q&A]
I have many contacts for each customer, so I have a one to many table relationship. How do log the changes to the contact while linking them to the customer?

Right now I have to do this to get the activity of the customer: 

        $activities = Activity::where('subject_id', $customer->id)
            ->where('log_name', 'customer')
            ->orderBy('created_at', 'desc')
            ->get();

How do I add the contacts to that?

**Comment by Gummibeer** [comment]
Hey,

I would recommend a new column in the activity table like `parent_type`&`parent_id` (polymorphic relation)  or whatever feels good for you.
But you could also do something like:

```php
Activity::query()
    ->where(function (\Illuminate\Database\Eloquent\Builder $query) use ($customer) {
        $query
            ->where(function (\Illuminate\Database\Eloquent\Builder $q) use ($customer) {
                $q
                    ->where('subject_type', $customer->getMorphClass())
                    ->where('subject_id', $customer->getKey());
            })
            ->orWhere(function (\Illuminate\Database\Eloquent\Builder $q) use ($customer) {
                $q
                    ->where('subject_type', (new Contact)->getMorphClass())
                    ->whereIn('subject_id', $customer->contacts()->pluck('id'));
            });
    })
    ->latest()
    ->get();
```

This query should return all activities related to the customer or any of his contacts.

**Reply by airliajsmith** [reply]
For the first option, how do I go about logging the extra field when activity occurs?
